Assets/Scripts/Character/CharacterControl.cs

```diff
@@ -39,12 +52,40 @@ public bool FireWeapon(Vector3 targetPosition)
         {
             return false;
         }
+
         rocketCooldownTimer = rocketCooldownDuration;
         var angle = Vector2.SignedAngle(Vector2.up, targetPosition - transform.position);
         var rotation = Quaternion.Euler(0.0f, 0.0f, angle);
         var rocket = Instantiate(rocketClass, transform.position, rotation);
         rocket.GetComponent<Rocket>().parentPlayerCollider = objectCollider;
         Physics2D.IgnoreCollision(objectCollider, rocket.GetComponent<Rocket>().objectCollider, true);
+
+        // spawn shoot_effect animation
+        GameObject shootEffectObject = Instantiate(shootEffect, transform.position + rotation * spriteOffset, rotation);
+        Destroy(shootEffectObject, .2f);
```

Comment:
How about activate and inactivate the GameObject instead of instantiate and destroy? But it's ok to merge first, and maybe we can do it later if we have time!